scratchpad/authentiacation.ts

```diff
@@ -0,0 +1,32 @@
+
+export class Authenticate {
+  constructor(
+    private @inject('authentication.required') required,
+    private @inject('authentication.user') user
+  )
+  static name = 'authentication.user';
+  value() {
+    console.log(this.user); // {username: ...};
+    return () => {
+        const isAuthRequired = this.required;
+        const user = this.user;
+
+        if (isAuthRequired && !user) {
+          throw ...;
+        }
+    }
+  }
+}
+
+// @loopback/core
+export class Invoke {
+  constructor(
+    private @inject('controller') controller,
+    private @inject('method') method,
+    private @inject('args') args,
+  )
+  static name = 'invoke';
```

Comment:
In our sync-up call, we mentioned that a static `name` may now work correctly, because we are effectively changing the reserved `.name` property of the constructor function. Was our assumption wrong? (FWIW, we were talking about using `key` property instead of `name`.)